### DISCUSSION TITLE
What happens to the github runners after it finishes?

### DISCUSSION THREAD

**Discussion by felixmosh** [community/community#26174] [Actions]
<p>I’m wondering what happens to the runner after it finishes it’s tasks? </p>
<p>Is it safe to assume that the files on it got deleted?</p>
<p>Is it safe to write some secrets to file for build purposes?</p>

**Comment by logankilpatrick** [comment, marked as answer]
<p>Hey <a class="mention" href="/felixmosh">@felixmosh</a>, the files on the runner are deleted after the workflow is done. If you have sensitive data, I suggest storing it in a GitHub secret (as to not expose it to others viewing the workflow file) but otherwise rest assured that there will not be any data persisting after your action is done so you do not need to worry about data being compromised. With that said, the level of caution and such you should take depends on what you are specifically doing so you may want to look into other forms of additional encryption for your data. </p>

**Comment by felixmosh** [comment]
<p>I’m saving my env vars  in secrets, but some of them I write to file during a workflow.</p>
<p>thank you for the answer.</p>
